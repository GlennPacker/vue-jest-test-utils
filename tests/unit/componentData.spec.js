import { mount } from '@vue/test-utils';
import HelloWorld from '@/components/HelloWorld.vue';

describe('Component Data -', () => {
    it('reading component data props', () => {
        // Arrange
        const wrapper = mount(HelloWorld)

        // Act
        const foo = wrapper.vm.foo

        // Assert
        expect(!!foo).toBe(true)
        expect(foo).toBe('hi')
    })


    it('setting component data props', async () => {
        // Arrange
        const wrapper = mount(HelloWorld);

        // Act
        wrapper.setData({ foo: 'bye'})

        // Assert
        await wrapper.vm.$nextTick()
        expect(wrapper.html()).toMatchSnapshot();

        const foo = wrapper.vm.foo;
        expect(!!foo).toBe(true)
        expect(foo).toBe('bye')
    })
})