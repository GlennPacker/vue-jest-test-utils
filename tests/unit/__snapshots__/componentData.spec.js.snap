// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Data - setting component data props 1`] = `
<div class="container">
  <h1>Hello </h1>
  <div>bye </div>
</div>
`;
